apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/schema/SchemaManager.java

```diff
@@ -410,8 +453,8 @@ private Uni<Schema.CqlKeyspaceDescribe> getKeyspaceInternal(
                     .flatMap(
                         keyspace -> {
                           // if it was not cached before, we can simply return
-                          // no need to check
-                          if (!cached) {
+                          // same if we don't want to validate hash
```

Comment:
Ok, logic here and above looks very complicated (and I know it already existed and is not changed here): first cache is accessed only to see if there is a cached entry and one that is not null (why would it be `null`? Do we cache "no such keyspace" entries?).
This produces `Boolean` (have in cache, true/false), instead of retaining Keyspace entry -- requiring separate cache lookup for some reason.
But then on line 407/450 we make call to `fetchKeyspace(ks, tenant, bridge)` which ... calls Bridge doesn't it?

So doesn't this still make actual call even if there's cached keyspace entry?

Reply:
Looking further into this, `fetchKeyspace()` apparently does indirectly access cache due to annotations.
So I guess it would avoid call if cache entry exists.

Reply:
Yes, unfortunately we do access cache twice.. That's because there is no way to know if the item was already in the cache or not.. And yes this can be improved :) Add a fix :+1:

Reply:
Fixed in https://github.com/stargate/stargate/pull/2355/commits/6494741439a60be3bb2244fb3885d5543e72959b

I think it was worth it:

* decrease cache access
* and actually fix that extra fetch in error case; as you can see in the commit in case of an error when calling the bridge, this error will also be cached.. however old implementation counted errors as `cached=true`, so next time we try we go fetch the keyspace (because last try was error) and plus validate hash (because error was cached)..